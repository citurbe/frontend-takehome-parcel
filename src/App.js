import React, { Component } from 'react';
import Header from './components/Header/Header';
import Search from './components/Search/Search';
import List from './components/List/List';
import gemApi from './api/index';
import { findSaved } from './util/findSaved';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: '',
      gemList: [],
      savedGems: []
    };
  }

  componentDidMount() {
    const loadFromStorage = localStorage.getItem('savedGems');
    if (loadFromStorage) {
      this.setState({ savedGems: JSON.parse(loadFromStorage) });
    }
  }

  handleChange(e) {
    this.setState({ searchText: e.target.value });
  }

  searchGems() {
    const queryString = this.state.searchText;
    gemApi.searchGems(queryString).then(resp => {
      this.setState({ gemList: resp.data });
    });
  }

  toggleSave(name) {
    const updated = findSaved(name, this.state.gemList, this.state.savedGems);
    this.setState({ savedGems: updated });
    localStorage.setItem('savedGems', JSON.stringify(updated));
  }

  showSaved() {
    this.setState({ gemList: this.state.savedGems, searchText: '' });
  }

  render() {
    return (
      <div>
        <Header />
        <div className="main-content">
          <Search
            handleChange={this.handleChange.bind(this)}
            searchText={this.state.searchText}
            handleSearch={this.searchGems.bind(this)}
            showSaved={this.showSaved.bind(this)}
          />
          <List
            toggleSave={this.toggleSave.bind(this)}
            gemList={this.state.gemList}
            savedGems={this.state.savedGems}
          />
        </div>
      </div>
    );
  }
}

export default App;
