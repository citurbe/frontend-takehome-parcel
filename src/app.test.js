import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

test('App display renders correctly', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
