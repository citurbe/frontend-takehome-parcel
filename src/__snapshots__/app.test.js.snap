// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App display renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h4>
      Ruby Gems Search
    </h4>
  </div>
  <div
    className="main-content"
  >
    <div>
      <input
        className="search-box"
        onChange={[Function]}
        type="text"
        value=""
      />
       
      <button
        className="search-button"
        onClick={[Function]}
      >
        Search
      </button>
      <button
        className="search-button"
        onClick={[Function]}
      >
        Show Saved
      </button>
    </div>
    <div />
  </div>
</div>
`;
